src/workerd/io/worker-fs.c++

```diff
@@ -0,0 +1,245 @@
+ public:
+  void setLastModified(jsg::Lock& js, kj::Date date = kj::UNIX_EPOCH) override {
+    KJ_IF_SOME(dir, tryGetDirectory()) {
+      dir.setLastModified(js, date);
+    }
+  }
+
+  const Stat stat() const override {
+    KJ_IF_SOME(dir, tryGetDirectory()) {
+      return dir.stat();
+    }
+    return Stat{
+      .type = Type::DIRECTORY,
+      .size = 0,
+      .lastModified = kj::UNIX_EPOCH,
+      .writable = true,
+    };
+  }
+
+  size_t count(jsg::Lock& js, kj::Maybe<Type> typeFilter = kj::none) const override {
+    KJ_IF_SOME(dir, tryGetDirectory()) {
+      return dir.count(js, typeFilter);
+    }
+    return 0;
+  }
+
+  void forEach(jsg::Lock& js,
+      kj::FunctionParam<bool(jsg::Lock& js, kj::StringPtr, Stat stat)> fn,
+      kj::Maybe<Type> typeFilter = kj::none) const override {
+    KJ_IF_SOME(dir, tryGetDirectory()) {
+      dir.forEach(js, fn, typeFilter);
+    }
+  }
+
+  kj::Maybe<kj::OneOf<kj::Own<jsg::fs::File>, kj::Own<jsg::fs::Directory>>> tryOpen(
+      jsg::Lock& js, kj::PathPtr path, kj::Maybe<Type> createAs = kj::none) override {
+    KJ_IF_SOME(dir, tryGetDirectory()) {
+      return dir.tryOpen(js, path, createAs);
+    }
+    return kj::none;
+  }
+
+  kj::Own<jsg::fs::File> transfer(
+      jsg::Lock& js, kj::StringPtr name, kj::Own<jsg::fs::File>&& file) override {
+    KJ_IF_SOME(dir, tryGetDirectory()) {
+      return dir.transfer(js, name, kj::mv(file));
+    }
+    JSG_FAIL_REQUIRE(Error, "Cannot transfer a file into a read-only directory");
+  }
+  kj::Own<Directory> transfer(
+      jsg::Lock& js, kj::StringPtr name, kj::Own<jsg::fs::Directory>&& dir) override {
+    KJ_IF_SOME(node, tryGetDirectory()) {
+      return node.transfer(js, name, kj::mv(dir));
+    }
+    JSG_FAIL_REQUIRE(Error, "Cannot transfer a directory into a read-only directory");
+  }
+
+  bool remove(
+      jsg::Lock& js, kj::PathPtr path, kj::Maybe<RemoveOptions> options = kj::none) override {
+    KJ_IF_SOME(dir, tryGetDirectory()) {
+      return dir.remove(js, path, options);
+    }
+    return false;
+  }
+
+  kj::StringPtr jsgGetMemoryName() const override {
+    return "TempDirectory"_kj;
+  }
+
+  size_t jsgGetMemorySelfSize() const override {
+    return sizeof(TempDirectory);
+  }
+
+  void jsgGetMemoryInfo(jsg::MemoryTracker& tracker) const override {
+    // The memory contents of this directory are not tracked. This is
+    // because they are entirely dependent on the current storage scope
+    // or IoContext. However, it is not likely that jsgGetMemoryInfo
+    // will be called when either of these are current.
+  }
+
+ private:
+  kj::Maybe<jsg::fs::Directory&> tryGetDirectory() const {
+    if (IoContext::hasCurrent()) {
+      return IoContext::current().getTempDirStoreScope().getDirectory();
+    }
+    if (TempDirStoreScope::hasCurrent()) {
+      return TempDirStoreScope::current().getDirectory();
+    }
+    return kj::none;
+  }
+};
+
+class LazyDirectory final: public jsg::fs::Directory {
```

Comment:
I couldn't find any documentation about what this class does? Can you add a comment explaining the difference, and possibly add a comment to directory class as well, mentioning the existence of this class?